tests/neg/i7618.scala

```diff
@@ -0,0 +1,36 @@
+package macros
+
+import scala.quoted.{given, _}
+
+enum Exp {
+  case Num(n: Int)
+  case Plus(e1: Exp, e2: Exp)
+  case Var(x: String)
+  case Let(x: String, e: Exp, in: Exp)
+}
+
+object Compiler {
+  import Exp._
+
+  inline def compile(e: Exp, env: Map[String, Expr[Int]])(given ctx: QuoteContext): Expr[Int] = inline e match { // error
```

Comment:
This may conflict with #7684. One of the PRs needs to be rebased on the other one.